Multiple errors while trying to install hpctoolkit with external compiler

### Steps to reproduce

I'm executing the command `spack install hpctoolkit~viewer~papi+rocm+mpi` and having several issues.

Here is my packages.yaml:
```yaml
packages:
  c:
    prefer:
    - 'llvm-amdgpu@=5.3.3'
  cxx:
    prefer:
    - 'llvm-amdgpu@=5.3.3'
  hip:
    externals:
    - spec: hip@5.3.3
      prefix: /opt/ohpc/pub/rocm/5.3.3
    buildable: false
  rocprofiler-dev:
    externals:
    - spec: rocprofiler-dev@5.3.3
      prefix: /opt/ohpc/pub/rocm/5.3.3
    buildable: false
  hsa-rocr-dev:
    externals:
    - spec: hsa-rocr-dev@5.3.3
      prefix: /opt/ohpc/pub/rocm/5.3.3
    buildable: false
  roctracer-dev:
    buildable: false
    externals:
    - spec: roctracer-dev@5.3.3
      prefix: /opt/ohpc/pub/rocm/5.3.3
  llvm-amdgpu:
    externals:
    - spec: llvm-amdgpu@=5.3.3
      prefix: /opt/ohpc/pub/rocm/5.3.3
      compilers:
        c: /opt/ohpc/pub/rocm/5.3.3/bin/amdclang
        cxx: /opt/ohpc/pub/rocm/5.3.3/bin/amdclang++
  gcc:
    externals:
    - spec: gcc@12.2.0 languages:='c,c++,fortran'
      prefix: /opt/ohpc/pub/compiler/gcc/12.2.0
      extra_attributes:
        compilers:
          c: /opt/ohpc/pub/compiler/gcc/12.2.0/bin/gcc
          cxx: /opt/ohpc/pub/compiler/gcc/12.2.0/bin/g++
          fortran: /opt/ohpc/pub/compiler/gcc/12.2.0/bin/gfortran
    - spec: gcc@7.5.0 languages:='c,c++,fortran'
      prefix: /usr
      extra_attributes:
        compilers:
          c: /usr/bin/gcc
          cxx: /usr/bin/g++
          fortran: /usr/bin/gfortran
  all:
    variants: amdgpu_target=gfx906
    target: [haswell]
  mpi:
    buildable: false
    require: openmpi@5.0.0
  openmpi:
    externals:
    - spec: openmpi@5.0.0
      #prefix: /opt/ohpc/pub/mpi/openmpi5-clang/5-mo49
      modules:
      - openmpi5/5-mo49
    buildable: false
```

### Error message

1. openssl
```
==> openssl: Executing phase: 'install'
==> Error: TypeError: argument of type 'NoneType' is not iterable

.spack/package_repos/fncqgg4/repos/spack_repo/builtin/packages/openssl/package.py:136, in install:
        133        if spec.satisfies("@1.0"):
        134            options.append("no-krb5")
        135        # clang does not support the .arch directive in assembly files.
  >>    136        if "clang" in self["c"].cc and spec.target.family == "aarch64":
        137            options.append("no-asm")
        138        elif "%nvhpc" in spec:
        139            # Last tested on nvidia@22.3 for x86_64:
```
2. boost
```
==> boost: Executing phase: 'install'
==> [2025-09-05-12:47:14.882970] './bootstrap.sh' '--prefix=/mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/boost-1.88.0-h2sl7gf4p2tocluu6nwu7xkcz3sv2sgy' '--with-toolset=gcc' '--with-libraries=atomic,chrono,container,date_time,exception,filesystem,graph,random,regex,system,thread,timer' '--without-icu'
Building B2 engine..

A C++11 capable compiler is required for building the B2 engine.
Toolset 'gcc' does not appear to support C++11.

> /mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/compiler-wrapper-1.0-c2nacljjc5uegakuwgcwuls776vb66oo/libexec/spack/rocmcc/amdclang++ -x c++ -std=c++11 -pthread  check_clib.cpp check_cxx11.cpp
/mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/compiler-wrapper-1.0-c2nacljjc5uegakuwgcwuls776vb66oo/libexec/spack/rocmcc/amdclang++: line 205: exec: None: not found
> /mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/compiler-wrapper-1.0-c2nacljjc5uegakuwgcwuls776vb66oo/libexec/spack/rocmcc/amdclang++ -x c++ -std=c++11  check_clib.cpp check_cxx11.cpp
/mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/compiler-wrapper-1.0-c2nacljjc5uegakuwgcwuls776vb66oo/libexec/spack/rocmcc/amdclang++: line 205: exec: None: not found
> /mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/compiler-wrapper-1.0-c2nacljjc5uegakuwgcwuls776vb66oo/libexec/spack/rocmcc/amdclang++ -x c++ -std=c++11 -D_GNU_SOURCE  check_clib.cpp check_cxx11.cpp
/mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/compiler-wrapper-1.0-c2nacljjc5uegakuwgcwuls776vb66oo/libexec/spack/rocmcc/amdclang++: line 205: exec: None: not found


** Note, the C++11 capable compiler is _only_ required for building the B2
** engine. The B2 build system allows for using any C++ level and any other
** supported language and resource in your projects.
```
3. libunwind, libxml2, libbsd have similar output:
```
  >> 20    configure: error: in '/tmp/vsgaligerov/spack-stage/spack-stage-libbsd-0.12.2-qo24m5sxt62wtarm5rsurcmp6iudvmsy/spack-src':
  >> 21    configure: error: C compiler cannot create executables
```
and log:
```
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether make supports the include directive... yes (GNU style)
checking for gcc... /mnt/beegfs/scratch/vsgaligerov/tools/spack/opt/spack/linux-haswell/compiler-wrapper-1.0-c2nacljjc5uegakuwgcwuls776vb66oo/libexec/spack/rocmcc/amdclang
checking whether the C compiler works... no
configure: error: in '/tmp/vsgaligerov/spack-stage/spack-stage-libbsd-0.12.2-qo24m5sxt62wtarm5rsurcmp6iudvmsy/spack-src':
configure: error: C compiler cannot create executables
See 'config.log' for more details
```

Before every error there is a warning:
```
==> Warning: /home/vsgaligerov/.spack/package_repos/fncqgg4/repos/spack_repo/builtin/packages/compiler_wrapper/package.py:174: when setting environment variable SPACK_CC=None: value is of type `NoneType`, but `str` was expected. This is deprecated and will be an error in Spack v1.0
==> Warning: /home/vsgaligerov/.spack/package_repos/fncqgg4/repos/spack_repo/builtin/packages/compiler_wrapper/package.py:174: when setting environment variable SPACK_CXX=None: value is of type `NoneType`, but `str` was expected. This is deprecated and will be an error in Spack v1.0
```


### Information on your system

* **Spack:** 1.1.0.dev0 (https://github.com/spack/spack/commit/d8c34f2b7a80c0927013408b126c386ff0f74cbe)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/d47ee47f97a6e6cabad5364f716e346cf64d0b15
* **Python:** 3.6.15
* **Platform:** linux-opensuse15-broadwell


### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have searched the issues of this repo and believe this is not a duplicate
- [x] I have run the failing commands in debug mode and reported the output

Before your suggestion I have tried to make some tools external because they also caused errors (perl, cmake, autoconf, automake). I have removed those externals, completely wiped spack and installed everything again. And it helped, now every dependency is installed successfully, except hpctoolkit itself. The latest version gives an error `>> 54    ../spack-src/meson.build:569:10: ERROR: Dependency "hip" not found, tried cmake`. Version 2024.01.1 gives  `  >> 3778    gpu/amd/rocprofiler-api.c:867:3: error: call to undeclared function 'hpcrun_force_dlopen'; ISO C99 and later do not support implicit function de
             clarations [-Wimplicit-function-declaration]`. And I succeeded only with version 2023.03.stable. Maybe this is some kind of a conflict that should be specified in the package config files...

--- a/lib/spack/spack/schema/packages.py
+++ b/lib/spack/spack/schema/packages.py
@@ -191,7 +191,7 @@ properties: Dict[str, Any] = {
                                 },
                             },
                         },
-                        "additionalProperties": True,
+                        "additionalProperties": False,
                         "required": ["spec"],
                     },
                 },

